ballerina/utils.bal

```diff
@@ -15,21 +15,147 @@
+    Uuid u = check toRecord(uuid);
+
+    int mostSigBits = u.timeLow & 0xffffffff;
+    mostSigBits <<= 16;
+    mostSigBits |= u.timeMid & 0xffff;
+    mostSigBits <<= 16;
+    mostSigBits |= u.timeHiAndVersion & 0xffff;
+
+    return mostSigBits;
+}
+
+isolated function getLeastSignificantBits(string uuid) returns int|Error {
+    Uuid u = check toRecord(uuid);
+
+    int leastSigBits;
+    string clockSeq = regex:split(uuid, "-")[3];
+    int|error clockSeqInt = int:fromHexString(clockSeq);
+    if clockSeqInt is int {
+        leastSigBits = clockSeqInt & 0xffff;
+    } else {
+        return error Error("Failed to get clock sequence value of the uuid");
+    }
+
+    leastSigBits <<= 48;
+    leastSigBits |= u.node & 0xffffffffffff;
+
+    return leastSigBits;
+}
+
+isolated function getUuidFromBytes(byte[] uuid) returns string {
+    int msb = ((uuid[0] & 0xFF) << 56) |
+            ((uuid[1] & 0xFF) << 48) |
+            ((uuid[2] & 0xFF) << 40) |
+            ((uuid[3] & 0xFF) << 32) |
+            ((uuid[4] & 0xFF) << 24) |
+            ((uuid[5] & 0xFF) << 16) |
+            ((uuid[6] & 0xFF) << 8) |
+            ((uuid[7] & 0xFF) << 0);
+
+    int lsb = ((uuid[8] & 0xFF) << 56) |
+            ((uuid[9] & 0xFF) << 48) |
+            ((uuid[10] & 0xFF) << 40) |
+            ((uuid[11] & 0xFF) << 32) |
+            ((uuid[12] & 0xFF) << 24) |
+            ((uuid[13] & 0xFF) << 16) |
+            ((uuid[14] & 0xFF) << 8) |
+            ((uuid[15] & 0xFF) << 0);
+
+    return getUuidFromSignificantBits(msb, lsb);
+}
+
+isolated function getBytesFromSignificantBits(int msb, int lsb) returns byte[] {
+    byte[] result = [];
+
+    result[0] = <byte>((msb >> 56) & 0xff);
+    result[1] = <byte>((msb >> 48) & 0xff);
+    result[2] = <byte>((msb >> 40) & 0xff);
+    result[3] = <byte>((msb >> 32) & 0xff);
+    result[4] = <byte>((msb >> 24) & 0xff);
+    result[5] = <byte>((msb >> 16) & 0xff);
+    result[6] = <byte>((msb >> 8) & 0xff);
+    result[7] = <byte>((msb >> 0) & 0xff);
+
+    result[8] = <byte>((lsb >> 56) & 0xff);
+    result[9] = <byte>((lsb >> 48) & 0xff);
+    result[10] = <byte>((lsb >> 40) & 0xff);
+    result[11] = <byte>((lsb >> 32) & 0xff);
+    result[12] = <byte>((lsb >> 24) & 0xff);
+    result[13] = <byte>((lsb >> 16) & 0xff);
+    result[14] = <byte>((lsb >> 8) & 0xff);
+    result[15] = <byte>((lsb >> 0) & 0xff);
+
+    return result;
+}
+
+isolated function getUuidFromSignificantBits(int mostSigBits, int leastSigBits) returns string {
+
+    byte[] bytes1 = formatUnsignedInt(leastSigBits, 4,12);
+    byte[] bytes2 = formatUnsignedInt(leastSigBits >>> 48, 4, 4);
+    byte[] bytes3 = formatUnsignedInt(mostSigBits, 4, 4);
+    byte[] bytes4 = formatUnsignedInt(mostSigBits >>> 16, 4, 4);
+    byte[] bytes5 = formatUnsignedInt(mostSigBits >>> 32, 4, 8);
+    
+    return checkpanic string:fromBytes(constructUuidBytesArray([bytes5,bytes4, bytes3, bytes2, bytes1]));
```

Comment:
```suggestion
    return checkpanic string:fromBytes(constructUuidBytesArray([bytes5, bytes4, bytes3, bytes2, bytes1]));
```

Reply:
Fixed in [b8d3323](https://github.com/ballerina-platform/module-ballerina-uuid/pull/376/commits/b8d33235fda972fc6799884b4de1a40aec5d18e4)